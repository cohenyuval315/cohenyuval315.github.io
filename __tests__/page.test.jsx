import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Page from "../src/app/page";

beforeEach(() => {
  // IntersectionObserver isn't available in test environment
  const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null
  });
  window.IntersectionObserver = mockIntersectionObserver;
});

describe("Page", () => {
  it("renders a heading", () => {
    render(<Page />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeInTheDocument();
  });
});

it("renders homepage unchanged", () => {
  const { container } = render(<Page />);
  expect(container).toMatchSnapshot();
});
