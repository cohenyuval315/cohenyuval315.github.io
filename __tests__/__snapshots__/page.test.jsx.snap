// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="bg-gray-900 text-white min-h-screen font-sans"
  >
    <div
      class="h-screen flex flex-col text-center bg-cover bg-center"
      style="opacity: 0; will-change: opacity;"
    >
      <div
        class="relative h-screen bg-cover bg-center"
        id="hero"
      >
        <div
          class="absolute top-0 left-0 w-full h-full"
        >
          <img
            alt="hero"
            data-nimg="fill"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=640&q=75 640w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=750&q=75 750w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=828&q=75 828w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fstatic%2Fimages%2Fhero.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; object-position: center; color: transparent; user-select: none; pointer-events: none; z-index: 0;"
          />
        </div>
        <div
          style="position: relative; z-index: 1;"
        >
          <section
            class="h-screen flex flex-col justify-center items-center text-center px-8 bg-cover bg-center"
            style="opacity: 0; will-change: opacity;"
          >
            <div
              style="user-select: none; pointer-events: none;"
            >
              <h1
                class="text-4xl md:text-6xl font-extrabold text-white mb-4"
                style="opacity: 0; will-change: transform,opacity; transform: translateX(-100px);"
              >
                Hi I'm Yuval Cohen
              </h1>
              <p
                class="text-lg md:text-xl mt-4 max-w-2xl text-gray-300"
                style="opacity: 0; will-change: transform,opacity; transform: translateY(50px);"
              >
                Software Engineer
              </p>
            </div>
            <div
              class="mt-6"
              style="opacity: 0; will-change: transform,opacity; transform: translateY(50px);"
            >
              <a
                class="bg-black text-white px-8 py-4 rounded-full flex items-center transform hover:scale-105 transition duration-300"
                href="https://github.com/cohenyuval315"
                target="_blank"
              >
                <svg
                  class="mr-2 text-white"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
                View My Work
              </a>
            </div>
          </section>
        </div>
      </div>
    </div>
    <section
      class="py-20 px-8 bg-blue-1000"
      style="opacity: 0;"
    >
      <div
        id="projects"
      >
        <div
          style="opacity: 0;"
        >
          <h2
            class="text-3xl font-extrabold mb-6 text-center text-white"
          >
            Projects
          </h2>
        </div>
        <section
          class="py-20 px-8"
          style="opacity: 0;"
        >
          <div
            class="grid gap-10 md:grid-cols-2 lg:grid-cols-3"
          >
            <div
              class=""
            >
              <div
                class="bg-gray-800 rounded-lg p-6 shadow-lg hover:shadow-2xl transition-all transform hover:scale-105"
              >
                <h3
                  class="text-xl font-semibold text-white"
                >
                  PyMicroservicesBase
                </h3>
                <p
                  class="text-gray-400 mt-2"
                />
                <div
                  class="mt-4 flex justify-between items-center"
                >
                  <a
                    class="text-blue-500 hover:underline"
                    href="https://github.com/cohenyuval315/PyMicroservicesBase"
                  >
                    GitHub
                  </a>
                </div>
              </div>
            </div>
            <div
              class=""
            >
              <div
                class="bg-gray-800 rounded-lg p-6 shadow-lg hover:shadow-2xl transition-all transform hover:scale-105"
              >
                <h3
                  class="text-xl font-semibold text-white"
                >
                  This Page
                </h3>
                <p
                  class="text-gray-400 mt-2"
                >
                  Simple Web Postfolio blasted with everything
                </p>
                <div
                  class="mt-4 flex justify-between items-center"
                >
                  <a
                    class="text-blue-500 hover:underline"
                    href="https://github.com/cohenyuval315/cohenyuval315.github.io"
                  >
                    GitHub
                  </a>
                  <a
                    class="text-blue-500 hover:underline"
                    href="https://cohenyuval315.github.io"
                    target="_blank"
                  >
                    Live Demo
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
    <section
      style="opacity: 0; transform: translateX(-100px);"
    >
      <section
        class="py-20 px-8 bg-gray-700"
        id="about"
      >
        <h2
          class="text-3xl font-extrabold mb-6 text-center text-white"
          style="opacity: 0; will-change: transform,opacity; transform: translateY(-50px);"
        >
          About Me
        </h2>
        <p
          class="text-lg max-w-3xl mx-auto text-center text-gray-300"
          style="opacity: 0; will-change: transform,opacity; transform: translateY(-50px);"
        >
          
    I’m a software engineer specializing in web development, python, and software design. I enjoy solving problems, learning new technologies, networking and connecting to others! .

        </p>
      </section>
    </section>
    <section
      class="py-20 px-8 bg-gray-900"
      style="opacity: 0;"
    >
      <section
        class="py-20 px-8 bg-gray-900"
        id="contact"
      >
        <h2
          class="text-3xl font-extrabold mb-6 text-center text-white"
        >
          CONTACT ME
        </h2>
        <p
          class="text-lg max-w-3xl mx-auto text-center mb-8 text-gray-300"
        >
          
    I’d love to hear from you! Whether you have a question, feedback, or just want to say hello, feel free to reach out via email. 
    You can also connect with me on GitHub and LinkedIn, or use the form below to send a message. 
    Let’s talk!

        </p>
        <div
          class="max-w-3xl mx-auto bg-gray-800 p-8 rounded-lg shadow-lg"
        >
          <form
            action="#"
            class="space-y-6"
            method="POST"
          >
            <div>
              <label
                class="block text-white text-lg"
                for="name"
              >
                Name
              </label>
              <input
                class="w-full px-4 py-2 bg-gray-700 text-white rounded-lg border border-gray-600 focus:outline-none focus:ring-2 focus:ring-blue-500"
                id="name"
                name="name"
                placeholder="Your Name"
                required=""
                type="text"
              />
            </div>
            <div>
              <label
                class="block text-white text-lg"
                for="email"
              >
                Email*
              </label>
              <input
                class="w-full px-4 py-2 bg-gray-700 text-white rounded-lg border border-gray-600 focus:outline-none focus:ring-2 focus:ring-blue-500"
                id="email"
                name="email"
                placeholder="Your Email"
                required=""
                type="email"
              />
            </div>
            <div>
              <label
                class="block text-white text-lg"
                for="message"
              >
                Message
              </label>
              <textarea
                class="w-full px-4 py-2 bg-gray-700 text-white rounded-lg border border-gray-600 focus:outline-none focus:ring-2 focus:ring-blue-500"
                id="message"
                name="message"
                placeholder="Your Message"
                required=""
                rows="4"
              />
            </div>
            <div>
              <label
                class="inline-flex items-center text-white"
              >
                <input
                  class="form-checkbox text-blue-500 focus:ring-2 focus:ring-blue-500"
                  required=""
                  type="checkbox"
                />
                <span
                  class="ml-2 text-sm"
                >
                  Sign up for our email list for updates, promotions, and more.
                </span>
              </label>
            </div>
            <div>
              <button
                class="w-full bg-blue-500 text-white py-3 rounded-lg shadow-md hover:bg-blue-600 transition duration-300"
                type="submit"
              >
                Send
              </button>
            </div>
          </form>
          <div
            class="mt-6 text-center text-sm text-gray-400"
          >
            <p>
              This site is protected by reCAPTCHA and the Google
               
              <a
                class="text-blue-500 hover:underline"
                href="https://policies.google.com/privacy"
                rel="noopener noreferrer"
                target="_blank"
              >
                Privacy Policy
              </a>
               
              and
               
              <a
                class="text-blue-500 hover:underline"
                href="https://policies.google.com/terms"
                rel="noopener noreferrer"
                target="_blank"
              >
                Terms of Service
              </a>
               
              apply.
            </p>
          </div>
        </div>
      </section>
      <div
        class="flex justify-center space-x-8 mt-10"
      >
        <div
          class="px-8 py-4 rounded-full flex items-center transform hover:scale-105 transition duration-300"
        >
          <a
            class="bg-black hover:bg-gray-800 text-white px-4 py-2 rounded-full flex items-center space-x-2 transition-transform transform hover:scale-105"
            href="https://github.com/cohenyuval315"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <span
              class="text-sm font-medium"
            >
              Github
            </span>
          </a>
        </div>
        <div
          class="px-8 py-4 rounded-full flex items-center transform hover:scale-105 transition duration-300"
        >
          <a
            class="bg-red-500 hover:bg-red-600 text-white px-4 py-2 rounded-full flex items-center space-x-2 transition-transform transform hover:scale-105"
            href="mailto:cohenyuval315@gmail.com""
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 488 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
              />
            </svg>
            <span
              class="text-sm font-medium"
            >
              Google
            </span>
          </a>
        </div>
        <div
          class="px-8 py-4 rounded-full flex items-center transform hover:scale-105 transition duration-300"
        >
          <a
            class="bg-blue-700 hover:bg-blue-800 text-white px-4 py-2 rounded-full flex items-center space-x-2 transition-transform transform hover:scale-105"
            href="https://www.linkedin.com/in/yc315/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              />
            </svg>
            <span
              class="text-sm font-medium"
            >
              Linkedin
            </span>
          </a>
        </div>
      </div>
    </section>
    <section>
      <script
        charset="UTF-8"
        crossorigin="*"
        src="https://embed.tawk.to/property_id/default"
      />
      <script>
        
                    var Tawk_API=Tawk_API||{}, Tawk_LoadStart=new Date();
                    (function(){
                        var s1=document.createElement("script"),s0=document.getElementsByTagName("script")[0];
                        s1.async=true;
                        s1.src='https://embed.tawk.to/679824ef825083258e0bcf55/1iil50i8a';
                        s1.charset='UTF-8';
                        s1.setAttribute('crossorigin','*');
                        s0.parentNode.insertBefore(s1,s0);
                    })();
                
      </script>
    </section>
  </div>
</div>
`;
